staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -183,8 +188,8 @@ func (npw *nestedPrefixWriter) WriteLine(a ...interface{}) {
 	npw.PrefixWriter.Write(npw.indent, "%s", fmt.Sprintln(a...))
 }
 
-func describerMap(clientConfig *rest.Config) (map[schema.GroupKind]ResourceDescriber, error) {
-	c, err := clientset.NewForConfig(clientConfig)
+func describerMap(clientConfig *rest.Config, client *http.Client) (map[schema.GroupKind]ResourceDescriber, error) {
+	c, err := clientset.NewForConfigAndClient(clientConfig, client)
```

Comment:
What if client is nil?

Maybe something like this?

```
if client == nil {
    c, err = clientset.NewForConfig(clientConfig)
} else {
    c, err = clientset.NewForConfigAndClient(clientConfig, client)
}
```

If not, then at a minimum it should probably return an error in `DescriberFor` if httpClient is nil.

Reply:
changed to this construct for safety